File: 2640-find-the-score-of-all-prefixes-of-an-array/2640-find-the-score-of-all-prefixes-of-an-array.js
/**
 * @param {number[]} nums
 * @return {number[]}
 */
var findPrefixScore = function(nums) {
    let max = 0;
    let prev = 0
    let res = []
    for(let i = 0; i < nums.length; i++){
        if(nums[i] > max){
            prev = max
            max = nums[i]
        }
        res.push(nums[i] + max)
        if(i > 0){
            res[i] = res[i] + res[i - 1]
        }
    }
    return res
};

File: 2640-find-the-score-of-all-prefixes-of-an-array/2640-find-the-score-of-all-prefixes-of-an-array.py
class Solution:
    def findPrefixScore(self, nums: List[int]) -> List[int]:
        max = 0
        res = []
        
        for x in range(len(nums)):
            if nums[x] > max:
                max = nums[x]
            
            res.append(nums[x] + max)
            if x > 0:
                res[x] = res[x] + res[x - 1]
                
        return res
    
    
    
